File: ml/src/main/scala/ai/h2o/sparkling/ml/params/HasInitialBiases.scala
package ai.h2o.sparkling.ml.params

import ai.h2o.sparkling.H2OFrame
import org.apache.spark.ml.linalg.DenseVector

trait HasInitialBiases extends H2OAlgoParamsBase {
  private val initialBiases = new NullableVectorArrayParam(
    this,
    "initialBiases",
    "A array of weight vectors to be used for bias initialization of every network layer." +
      "If this parameter is set, the parameter 'initialWeights' has to be set as well.")

  setDefault(initialBiases -> null)

  def getInitialBiases(): Array[DenseVector] = $(initialBiases)

  def setInitialBiases(value: Array[DenseVector]): this.type = set(initialBiases, value)

  private[sparkling] def getInitialBiasesParam(trainingFrame: H2OFrame): Map[String, Any] = {
    Map("initial_biases" -> convertVectorArrayToH2OFrameKeyArray(getInitialBiases()))
  }

  override private[sparkling] def getSWtoH2OParamNameMap(): Map[String, String] = {
    super.getSWtoH2OParamNameMap() ++ Map("initialBiases" -> "initial_biases")
  }
}

File: py-scoring/src/ai/h2o/sparkling/ml/params/HasInitialBiases.py
from ai.h2o.sparkling.ml.params.H2OTypeConverters import H2OTypeConverters
from pyspark.ml.param import *


class HasInitialBiases(Params):
    initialBiases = Param(
        Params._dummy(),
        "initialBiases",
        "A array of weight vectors to be used for bias initialization of every network layer. "
        "If this parameter is set, the parameter 'initialWeights' has to be set as well.",
        H2OTypeConverters.toNullableListDenseVector())

    def getInitialBiases(self):
        return self.getOrDefault(self.initialBiases)

    def setInitialBiases(self, value):
        return self._set(initialBiases=value)
